### PR TITLE
Creates "Connect to Google vertex" llm connector guide

### LINKED ISSUE #5373: [Request] New Google Gemini Connector for GenAI
### Description

We are introducing a new Gemini connector, which will enable our customers to leverage the Gemini family of models (i.e. ultra, pro, flash, nano) with Elastic Security GenAI features like the AI Assistant and Attack Discovery.

### Background & resources

* PRs: 
https://github.com/elastic/kibana/pull/184741
https://github.com/elastic/kibana/pull/183668

* Point of contact: 
@stephmilovic 

* Test environments: 
https://kibana-pr-184741.kb.us-west2.gcp.elastic-cloud.com:9243/

### Which documentation set does this change impact?

ESS and serverless

### ESS release

8.15

### Serverless release

The week of June 17, 2024

### Feature differences

The feature is identical in ESS and Serverless.

### API docs impact

- We will need to update the main [AI Assistant page](https://www.elastic.co/guide/en/security/8.14/security-assistant.html), where we mention connectors.
- We should also add a detailed setup connector guide for Gemini and reference it [in this page](https://www.elastic.co/guide/en/security/8.14/llm-connector-guides.html).
- We should update the [Large language model performance matrix](https://www.elastic.co/guide/en/security/8.14/llm-performance-matrix.html) page.

### Prerequisites, privileges, feature flags

No feature flag used.

### LINKED ISSUE #5459: [Request] Connect Elastic AI Assistant to Google Vertex
### Description

Publish documentation for integrating Elastic with Google Vertex: https://docs.google.com/document/d/1DoVPMc5FxVKnojcD8WVLQG12m6K4NzT5tSbRhwMV0Cw/edit?usp=sharing

Documentation is finished along with videos uploaded to Vidyard. Please put a link under the main landing page for [Setting up connectors for LLM](https://www.elastic.co/guide/en/security/current/llm-connector-guides.html). 

[Set up connectors for large language models (LLM)]
[Connect to Azure OpenAI]
[Connect to OpenAI]
[Connect to Amazon Bedrock]
[Connect to Google Vertex]

### Background & resources

Deploy to documents: https://docs.google.com/document/d/1DoVPMc5FxVKnojcD8WVLQG12m6K4NzT5tSbRhwMV0Cw/edit?usp=sharing

### Which documentation set does this change impact?

ESS and serverless

### ESS release

8.14

### Serverless release

Next week, 24 July

### Feature differences

N/A

### API docs impact

N/A

### Prerequisites, privileges, feature flags

N/A

### PR DESCRIPTION
**Pull request by benironside**
Fixes #5459 and fixes #5373. Creates a new LLM connector guide for Google Vertex AI.

Preview: [Google Vertex AI ](https://elastic-dot-co-docs-production-elasticdocs-elastic-dev.vercel.app/current/serverless/security/connect-to-google-vertex)

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-r7ullcysq-elastic-dev.vercel.app

* Built with commit 50eb0dc4c4a2e6eb588f8039e523a9bbb50ede4c

> Issues? Visit #next-docs in Slack

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-jv9z0q0wa-elastic-dev.vercel.app

* Built with commit 50eb0dc4c4a2e6eb588f8039e523a9bbb50ede4c

> Issues? Visit #next-docs in Slack

File: docs/serverless/assistant/connect-to-vertex.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+<DocVideo source="vidyard" id="hrcy3F9AodwhJcV1i2yqbG"/>
+
+## Configure the Google Gemini connector
+
+Finally, configure the connector in ((kib)):
+
+1. Log in to ((kib)).
+2. Navigate to **Stack Management → Connectors → Create Connector → Google Gemini**.
+3. Name your connector to help keep track of the model version you are using.
+4. The **URL** field can be left as default.
```
Comment (Charelzard) [inline docs/serverless/assistant/connect-to-vertex.mdx:60]: Change to: Enter URL based on your region url.

File: docs/serverless/assistant/connect-to-vertex.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+5. Enter your **GCP Region** and **GCP Project ID**.
+6. Under **Default model**, specify which [model](https://cloud.google.com/vertex-ai/generative-ai/docs/learn/models) you want to use.
```
Comment (Charelzard) [inline docs/serverless/assistant/connect-to-vertex.mdx:62]: Change to: Under Default Model, specify either:
gemini-1.5-pro
Gemini-1.5-flash

Note: for further information on model differences go [here](https://cloud.google.com/vertex-ai/generative-ai/docs/learn/models).

**Review by Charelzard — COMMENTED** [review]
Please see comments listed.

File: docs/serverless/assistant/connect-to-vertex.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+---
+slug: /serverless/security/connect-to-google-vertex
+title: Connect to Google Vertex
+description: Set up a Google Vertex LLM connector.
+tags: ["security", "overview", "get-started"]
+status: in review
+---
+
+# Connect to Google Vertex AI
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-vertex.mdx:9]: This line makes it look like there are two H1s on the page – **Connect to Google Vertex** and **Connect to Google Vertex AI**. We can probably remove this line.

File: docs/serverless/assistant/connect-to-vertex.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+This page provides step-by-step instructions for setting up a Google Vertex AI connector for the first time. This connector type enables you to leverage Vertex AI's large language models (LLMs) within ((kib)). You'll first need to enable Vertex AI, then generate an API key, and finally configure the connector in ((kib)).
+
+<DocCallOut title="Important"> 
+Before continuing, you should have an active project in one of Google Vertex AI's [supported regions](https://cloud.google.com/vertex-ai/docs/general/locations#feature-availability). 
+</DocCallOut>
+
+## Emable the Vertex AI API
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-vertex.mdx:17]:
```suggestion
## Enable the Vertex AI API
```

File: docs/serverless/assistant/connect-to-vertex.mdx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+1. Log in to the GCP console and navigate to **Vertex AI → Vertex AI Studio → Overview**. 
+2. If you're new to Vertex AI, the **Get started with Vertex AI Studio** popup appears. Click **Vertex AI API**, then click **ENABLE**.
+
+The following video demonstrates these steps.
+
+<DocVideo source="vidyard" id="vFhtbiCZiKhvdZGy2FjyeT"/>
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-vertex.mdx:24]: Not sure why, but all videos on the page render as screenshots.

File: docs/serverless/assistant/connect-to-vertex.mdx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+For more information about enabling the Vertex AI API, refer to [Google's documentation](https://cloud.google.com/vertex-ai/docs/start/cloud-environment).
+
+
+## Create a Vertex AI service account
+
+1. In the GCP console, navigate to **APIs & Services → Library**.
+2. Search for **Vertex AI API**, select it, and click **MANAGE**.
+3. In the left menu, navigate to **Credentials** then click **+ CREATE CREDENTIALS** and select **Service account**.
+4. Name the new service account then click **CREATE AND CONTINUE**.
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-vertex.mdx:34]:
```suggestion
4. Name the new service account, then click **CREATE AND CONTINUE**. 
```

File: docs/serverless/assistant/connect-to-vertex.mdx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+slug: /serverless/security/connect-to-google-vertex
+title: Connect to Google Vertex
+description: Set up a Google Vertex LLM connector.
+tags: ["security", "overview", "get-started"]
+status: in review
+---
+
+# Connect to Google Vertex AI
+
+This page provides step-by-step instructions for setting up a Google Vertex AI connector for the first time. This connector type enables you to leverage Vertex AI's large language models (LLMs) within ((kib)). You'll first need to enable Vertex AI, then generate an API key, and finally configure the connector in ((kib)).
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-vertex.mdx:11]:
```suggestion
This page provides step-by-step instructions for setting up a Google Vertex AI connector for the first time. This connector type enables you to leverage Vertex AI's large language models (LLMs) within the ((security-app)). You'll first need to enable Vertex AI, then generate an API key, and finally configure the connector in your Security project.
```

File: docs/serverless/assistant/connect-to-vertex.mdx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+3. Go to the **KEYS** tab, click **ADD KEY**, then select **Create new key**.
+4. Select **JSON**, then click **CREATE** to download the key. Keep it somewhere secure.
+
+The following video demonstrates these steps.
+
+<DocVideo source="vidyard" id="hrcy3F9AodwhJcV1i2yqbG"/>
+
+## Configure the Google Gemini connector
+
+Finally, configure the connector in ((kib)):
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-vertex.mdx:55]:
```suggestion
Finally, configure the connector in your Security project:
```

File: docs/serverless/assistant/connect-to-vertex.mdx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+1. Log in to ((kib)).
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-vertex.mdx:57]:
```suggestion
1. Log in to your Security project.
```

File: docs/serverless/assistant/connect-to-vertex.mdx (hunk 5 above)
Comment (benironside) [reply docs/serverless/assistant/connect-to-vertex.mdx:24]: They render correctly if you refresh, in my experience 🤷

File: docs/serverless/assistant/connect-to-vertex.mdx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+## Enable the Vertex AI API
+
+1. Log in to the GCP console and navigate to **Vertex AI → Vertex AI Studio → Overview**. 
+2. If you're new to Vertex AI, the **Get started with Vertex AI Studio** popup appears. Click **Vertex AI API**, then click **ENABLE**.
+
+The following video demonstrates these steps.
+
+<DocVideo source="vidyard" id="vFhtbiCZiKhvdZGy2FjyeT"/>
+
+For more information about enabling the Vertex AI API, refer to [Google's documentation](https://cloud.google.com/vertex-ai/docs/start/cloud-environment).
```
Comment (nastasha-solomon) [inline docs/serverless/assistant/connect-to-vertex.mdx:24]: I almost missed this line while reviewing the preview. If you wanted to highlight it more, maybe consider moving it to a note or a tip.

**Review by nastasha-solomon — DISMISSED** [review]

**Review by nastasha-solomon — APPROVED** [review]

**Pull request merged**